k-distribution/tests/pyk/emit-json-spec-tests/emit_json_spec_test.py

```diff
@@ -0,0 +1,83 @@
+import json
+import os
+import shutil
+import unittest
+from pathlib import Path
+
+from pyk.kast import KApply, KClaim, KDefinition, KRequire, KVariable
+from pyk.kastManip import rewriteAnywhereWith
+from pyk.ktool import KProve
+
+
+class EmitJsonSpecTest(unittest.TestCase):
+    TEST_DIR = 'emit-json-spec-tests'
+    MAIN_FILE = 'verification.k'
+    USE_DIR = f'{TEST_DIR}/.kprove'
+
+    def setUp(self):
+        shutil.rmtree(self.USE_DIR, ignore_errors=True)
+        os.makedirs(self.USE_DIR)
+
+        self.kprove = KProve(f'{self.TEST_DIR}/verification-kompiled', self.MAIN_FILE, self.USE_DIR)
+        self.kprove.proverArgs += ['-I', '.']
+        self._update_symbol_table(self.kprove)
+
+        with open(f'{self.TEST_DIR}/looping-spec.json', 'r') as spec_file:
+            module = json.load(spec_file)['term']
+
+        claim = extract_claims(module)[0]
+        self.claim = KClaim(
+            body=eliminate_generated_top(claim['body']),
+            requires=claim['requires'],
+            ensures=claim['ensures'],
+            att=None
+        )
+
+        module['localSentences'] = [self.claim]
+        self.module = module
+
+    def tearDown(self):
+        shutil.rmtree(self.USE_DIR)
+
+    @staticmethod
+    def _update_symbol_table(kprove):
+        def paren(f):
+            def unparse(*args):
+                return '(' + f(*args) + ')'
+            return unparse
+
+        kprove.symbolTable['_+Int_'] = paren(kprove.symbolTable['_+Int_'])
+
+    def test_prove_claim(self):
+        # When
+        result = self.kprove.proveClaim(self.claim, 'looping-1')
+
+        # Then
+        self.assertEqual(result['label'], '#Top')
+
+    def test_prove(self):
+        # Given
+        spec_name = 'looping-2-spec'
+        spec_path = Path(f'{self.USE_DIR}/{spec_name}.k')
+        spec_module_name = spec_name.upper()
+
+        self.module['name'] = spec_module_name
+        definition = KDefinition(self.module, [self.module], requires=[KRequire(self.MAIN_FILE)])
```

Comment:
Maybe `--emit-json-spec` could return the module in a `KDefinition` with requires (it is easier to extract the module if the definition is not needed than to reconstruct the definition when it is). What do you think, @ehildenb?

Reply:
It's actually not clear to me that it makes sense to do so. The "Definition" associated with the `specModule` is not super well defined, and is also not directly computed by the current frontend. I would basically have 2 choices:

- A definition with _just_ the `specModule` in it, so I'd be filling in sensible dummy parameters for it like you do here in pyk, so that the requires is set up correctly basically. `specModule` becomes the `mainModule` of this definition, which isn't very sensible.
- Assume that the user wants the entire kompiled definition, and not just a dummy definition with the single module in it. Once again the `specModule` becomes the `mainModule` of this definition, which isn't very sensible.

So I'm inclined to say "emitting just the module is more flexible, and it's up to the user to determine how to use that info". Since it's enough for now to accomplish our goals, I think we should wait and see if more info/structure than that is needed.